cluster_test.go

```diff
@@ -821,89 +837,9 @@ func TestClusterMin(t *testing.T) {
 	}
 }
 
-func (ts *testSequins) getShardID(t *testing.T) string {
-	url := fmt.Sprintf("http://%s/healthcheck", ts.name)
-
-	resp, err := ts.testClient.Get(url)
-	assert.NoError(t, err)
-	defer resp.Body.Close()
-
-	return resp.Header.Get("X-Sequins-Shard-ID")
-}
-
-func TestClusterAutoAssignSingleID(t *testing.T) {
-	if testing.Short() {
-		t.Skip("skipping cluster test in short mode.")
-	}
-	t.Parallel()
-
-	tc := newTestCluster(t)
-	defer tc.tearDown()
-
-	s1 := tc.addSequins(noShardID())
-	tc.makeVersionAvailable(v3)
-	tc.startTest()
-	time.Sleep(expectTimeout)
-
-	shardID := s1.getShardID(t)
-	assert.Equal(t, "1", shardID)
-}
-
-// Test shardID auto-assignment when spinning up nodes one at a time.
-func TestClusterAutoAssignMultipleIDs(t *testing.T) {
-	if testing.Short() {
-		t.Skip("skipping cluster test in short mode.")
-	}
-	t.Parallel()
-
-	tc := newTestCluster(t)
-	defer tc.tearDown()
-
-	for i := 1; i <= 3; i++ {
-		s := tc.addSequins(noShardID())
-		s.makeVersionAvailable(v3)
-		s.startTest()
-		time.Sleep(expectTimeout)
-	}
-
-	for i, s := range tc.sequinses {
-		shardID := fmt.Sprintf("%d", i+1)
-		assert.Equal(t, shardID, s.getShardID(t))
-	}
-}
-
-// Test shardID auto-assignment when the nodes start at the same time.
-func TestClusterAutoAssignParallelIDs(t *testing.T) {
-	if testing.Short() {
-		t.Skip("skipping cluster test in short mode.")
-	}
-	t.Parallel()
-
-	tc := newTestCluster(t)
-	defer tc.tearDown()
-
-	tc.addSequinses(3, noShardID())
-	tc.makeVersionAvailable(v3)
-	tc.startTest()
-	time.Sleep(expectTimeout * 3)
-
-	for i := 1; i <= 3; i++ {
-		shardID := fmt.Sprintf("%d", i)
-
-		found := false
-		for _, s := range tc.sequinses {
-			if s.getShardID(t) == shardID {
-				found = true
-				break
-			}
-		}
-		assert.True(t, found, "None of the nodes had shardID %q", shardID)
-	}
-}
-
-// Down each node in a cluster one at a time and then ensure shardID auto-assignment will use
-// the newly unused shardID.
-func TestClusterAutoAssignUnusedID(t *testing.T) {
+// TestClusterPartitionAssignment tests that the partitions are assigned to
+// nodes as expected since the assignments should be deterministic.
+func TestClusterPartitionAssignment(t *testing.T) {
```

Comment:
This only covers a very basic case. I'm currently trying to test the flow of:
1. spin up 5 nodes, ensure they all have expected partitions
2. kill a few nodes, ensure that existing nodes remain
3. spin up nodes to replace old nodes, since they won't have same hostnames, ensure that they have partitions based on new ordering
4. make new version available, ensure all partitions become uniformly distributed again

Reply:
Oh disregard this, I'm totally mistaken